corehq/form_processor/utils/general.py

```diff
@@ -1,16 +1,25 @@
 from django.conf import settings
 
 from corehq.apps.domain.models import Domain
-from corehq.toggles import USE_SQL_BACKEND
+from corehq.toggles import USE_SQL_BACKEND, NAMESPACE_DOMAIN
 
 
-def should_use_sql_backend(domain):
+def should_use_sql_backend(domain_name):
     if settings.UNIT_TESTING:
         override = getattr(settings, 'TESTS_SHOULD_USE_SQL_BACKEND', None)
         if override is not None:
             return override
 
-    return USE_SQL_BACKEND.enabled(domain) or Domain.get_by_name(domain).use_sql_backend
+    toggle_enabled = USE_SQL_BACKEND.enabled(domain_name)
+    if toggle_enabled:
+        domain = Domain.get_by_name(domain_name)
+        if not domain.use_sql_backend:
+            domain.use_sql_backend = True
+            domain.save()
+            USE_SQL_BACKEND.set(domain_name, enabled=False, namespace=NAMESPACE_DOMAIN)
+        return True
+
+    return toggle_enabled or Domain.get_by_name(domain_name).use_sql_backend
```

Comment:
another thought (and this is maybe premature optimization) would be to allow passing in a domain object or name so that you don't have to get it from the db or cache again if you already have it.